data/org.freedesktop.portal.Usb.xml

```diff
@@ -0,0 +1,222 @@
+
+        * ``parent`` (``s``)
+
+          Device ID of the parent device.
+
+        * ``readable`` (``b``)
+
+          Whether the device can be opened for reading with
+          org.freedesktop.portal.Usb.AcquireDevices(). If not present, then
+          it should be assumed to be false.
+
+        * ``writable`` (``b``)
+
+          Whether the device can be opened for writing with
+          org.freedesktop.portal.Usb.AcquireDevices(). If not present, then
+          it should be assumed to be false.
+
+        * ``device-file`` (``s``)
+
+          A string path to the device node inside the /dev filesystem.
+
+        * ``properties`` (``a{sv}``)
+
+          A list of udev properties that this device has. These properties
+          are not parsed in any way by the portal, it is up to apps to parse
+          them.
+     -->
+    <method name="EnumerateDevices">
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="a(sa{sv})" name="devices" direction="out"/>
+    </method>
+
+    <!--
+        AcquireDevices:
+        @parent_window: Identifier for the application window, see :ref:`Common Conventions <window-identifiers>`.
+        @devices: Array of device identifiers, device information, and access options
+        @options: Vardict with optional further information
+        @handle: Object path for the :ref:`org.freedesktop.portal.Request` object representing this call
+
+        Acquires the given device nodes.
+
+        Each element of the @devices array contains the device ID, and the
+        following keys:
+
+        * ``writable`` (``b``)
+
+          Whether the device will be opened in read-write or read-only mode.
+          Default: False
+
+        Supported keys in the @options vardict include:
+
+        * ``handle_token`` (``s``)
+
+          A string that will be used as the last element of the @handle. Must be a valid
+          object path element. See the :ref:`org.freedesktop.portal.Request` documentation
+          for more information about the @handle.
+
+        The #org.freedesktop.portal.Request::Response signal is emitted without
+        any extra information.
+     -->
+    <method name="AcquireDevices">
+      <arg type="s" name="parent_window" direction="in"/>
+      <arg type="a(sa{sv})" name="devices" direction="in"/>
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="o" name="handle" direction="out"/>
+    </method>
+
+    <!--
+        FinishAcquireDevices:
+        @options: Vardict with optional further information
+        @results: Array of device ids, and the result of the access
+        @finished: Whether all device results were reported
+
+        Retrieves the file descriptors of the devices requested during
+        org.freedesktop.portal.Usb.AcquireDevices().
+
+        This method can only be called once, and only after calling
+        org.freedesktop.portal.Usb.AcquireDevices().
```

Comment:
Should mention the `#org.freedesktop.portal.Request::Response` signal

Reply:
I think the response signal should carry an id that should be passed to `FinishAcquireDevices`. That way there is no problem with calling `AcquireDevices` multiple times before calling `FinishAcquireDevices`.